src/module_4/module_4_model-fitting.md

```diff
@@ -0,0 +1,1214 @@
+test_order_ids = order_ids[val_end:]
+
+train_df = df_filtered[df_filtered['order_id'].isin(train_order_ids)]
+val_df = df_feature_frame[df_feature_frame['order_id'].isin(val_order_ids)]
+test_df = df_feature_frame[df_feature_frame['order_id'].isin(test_order_ids)]
+
+print(f"Train set size: {len(train_df)}")
+print(f"Validation set size: {len(val_df)}")
+print(f"Test set size: {len(test_df)}")
+```
+
+    Train set size: 1446691
+    Validation set size: 347233
+    Test set size: 370029
+
+
+
+```python
+def feature_label_split(df, label_col):
+    return df.drop(label_col, axis=1), df[label_col]
+
+X_train, y_train = feature_label_split(train_df, label_col)
+X_val, y_val = feature_label_split(val_df, label_col)
+X_test, y_test = feature_label_split(test_df, label_col)
+```
+
+
+```python
+"""""Función para crear los gráficos del performance de nuestro modelo"""
+def plot_metrics(model_name, y_pred, y_test, figsize=(10, 7)):
+    precision, recall, _ = precision_recall_curve(y_test, y_pred)
+    pr_auc = auc(recall, precision)
+    fpr, tpr, _ = roc_curve(y_test, y_pred)
+    roc_auc = roc_auc_score(y_test, y_pred)
+    accuracy = accuracy_score(y_test.astype(np.int64), np.rint(y_pred).astype(np.int64))
+    conf_matrix = confusion_matrix(y_test.astype(np.int64), np.rint(y_pred).astype(np.int64))
+
+    fig, ax = plt.subplots(2, 2, figsize=figsize)  # 2x2 para incluir la matriz de confusión
+
+    ax[0, 0].plot(recall, precision, label=f"{model_name} AUC: {pr_auc:.2f}")
+    ax[0, 0].set_xlabel("Recall")
+    ax[0, 0].set_ylabel("Precision")
+    ax[0, 0].set_title(f"Precision-Recall Curve")
+    ax[0, 0].legend()
+
+    ax[0, 1].plot(fpr, tpr, label=f"AUC: {roc_auc:.2f}")
+    ax[0, 1].set_xlabel("FPR")
+    ax[0, 1].set_ylabel("TPR")
+    ax[0, 1].set_title(f"ROC Curve")
+    ax[0, 1].legend()
+
+    ax[1, 0].axis('off')  # No se necesita eje para mostrar texto
+    ax[1, 0].text(0.5, 0.5, f"Accuracy: {accuracy:.2f}", fontsize=15, ha='center')
+
+    sns.heatmap(conf_matrix, annot=True, fmt='d', cmap='Blues', ax=ax[1, 1])
+    ax[1, 1].set_title("Confusion Matrix")
+    ax[1, 1].set_xlabel("Predicted label")
+    ax[1, 1].set_ylabel("True label")
+
+    plt.tight_layout()
+    plt.show()
+    print("Precision Recall AUC: " + str(pr_auc))
+    print("Roc Curve AUC: " + str(roc_auc))
+    print("Sum of both AUCs: " + str(pr_auc + roc_auc))
+```
+
+
+```python
+"""""Función para ver la importancia de los coeficientes"""
+def plot_feature_importance(clf, columns):
+    fig, ax = plt.subplots()
+    fi = pd.DataFrame(list(zip(columns, clf.feature_importances_)), columns=["features", "importance"])
+    fi.sort_values(by="importance", ascending=True, inplace=True)
+    fi.plot(kind="barh", x="features", y="importance", ax=ax)
+    return fig, ax
+```
+
+## Previous Baselines
+
+### Initial Benchmark 
+
+Para nuestra evaluación inicial, utilizamos la característica de "popularidad global" para determinar si el pedido puede ser realizado o no.
```

Comment:
how is this global popularity computed? if it's taking into account orders of the WHOLE period then you are leaking data and shouldn't be used